File: 0179 Largest Number/0179.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> v;
        
        for (const int& num: nums){
            v.push_back(to_string(num));
        }
        
        sort(v.begin(), v.end(), [](const string& a, const string& b){
            return a+b > b+a;
        });

        if (v[0] == "0") return "0";

        string ans;

        for (const string& s: v){
            ans += s;
        }

        return ans;
    }
};

File: 0179 Largest Number/0179.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        nums = [str(x) for x in nums]
        
        def cmp(a, b):
            return 1 if a+b <= b+a else -1

        nums.sort(key = cmp_to_key(cmp))
        ans = "".join(nums)
        return "0" if ans[0] == '0' else ans
